docs/security.md

```diff
@@ -133,3 +133,174 @@ spec:
       cpu: "1"
       memory: "1512Mi"
 ```
+
+### Kubernetes API Access
+
+
+ By default, operator configures Kubernetes API Access for all managed components with own `ServiceAccount`.
+This behaviour can be altered with object configuration - `spec.disableAutomountServiceAccountToken: true`. See the
+following [Kubernetes doc](https://kubernetes.io/docs/tasks/configure-pod-container/configure-service-account/#opt-out-of-api-credential-automounting) for details.
+
+ Consider the following example for VMAgent:
+```yaml
+apiVersion: operator.victoriametrics.com/v1beta1
+kind: VMAgent
+metadata:
+  name: example-vmagent
+  namespace: default
+spec:
+  remoteWrite:
+  - url: http://vmsingle-vms-victoria-metrics-k8s-stack.default.svc:8429/api/v1/write
+  replicaCount: 1
+  selectAllByDefault: true
+  statefulMode: true
+```
+
+ Kubernetes controller-manager creates the following `Pod` definition and attaches `volumes` and `volumeMounts` with serviceAccount token:
+
+```yaml
+apiVersion: v1
+kind: Pod
+metadata:
+  name: vmagent-example-vmagent-0
+  namespace: default
+spec:
+  containers:
+  - args:
+...
+    name: config-reloader
+    volumeMounts:
+...
+    - mountPath: /var/run/secrets/kubernetes.io/serviceaccount
+      name: kube-api-access-q44gh
+      readOnly: true
+  - args:
+...
+    name: vmagent
+    volumeMounts:
+...
+    - mountPath: /var/run/secrets/kubernetes.io/serviceaccount
+      name: kube-api-access-q44gh
+      readOnly: true
+  dnsPolicy: ClusterFirst
+  enableServiceLinks: true
+  hostname: vmagent-example-vmagent-0
+  initContainers:
+  - args:
+...
+    name: config-init
+    volumeMounts:
+...
+    - mountPath: /var/run/secrets/kubernetes.io/serviceaccount
+      name: kube-api-access-q44gh
+      readOnly: true
+
+  serviceAccount: vmagent-example-vmagent
+  serviceAccountName: vmagent-example-vmagent
+  volumes:
+...
+  - emptyDir: {}
+    name: persistent-queue-data
+  - name: kube-api-access-q44gh
+    projected:
+      defaultMode: 420
+      sources:
+      - serviceAccountToken:
+          expirationSeconds: 3607
+          path: token
+      - configMap:
+          items:
+          - key: ca.crt
+            path: ca.crt
+          name: kube-root-ca.crt
+      - downwardAPI:
+          items:
+          - fieldRef:
+              apiVersion: v1
+              fieldPath: metadata.namespace
+            path: namespace
+```
+
+ If `disableAutomountServiceAccountToken: true` is set. Operator adds `volumes` and `volumeMounts` only if application explicitly requires access to Kubernetes API:
```

Comment:
>only if application explicitly requires access to Kubernetes API:

Should we clarify more about the applications here only include vmagent and vm config-reloader? Other components like vmcluster require users to mount it manually, can we also provide an example for that?

Reply:
It's a good point. And it only makes sense if there is some side-car components, that require access to the Kubernetes API.